kv/dist_sender.go

```diff
@@ -565,20 +571,21 @@ func (ds *DistSender) Send(_ context.Context, call client.Call) {
 			}
 
 			// For an RPC error to occur, we've must've been unable to contact
-			// any replicas. In this case, likely all nodes are down; this
-			// should really only happen if either the replica set is dead or
+			// any replicas. In this case, either
+			// - we sent only to a leader or
+			// - all nodes are down.
+			// The second case only happens if either the replica set is dead or
 			// we have an old descriptor which does not have a list of current
 			// nodes; we hope it's the latter.
-			// TODO(tschottdorf): individual rpc calls always return an
-			// rpcError, which is always retriable. Thus, rpc.Send will always
-			// return retriable errors if everything is unreachable. That seems
-			// awkward; rpcError should probably not be retriable (rpc.Send
-			// will still succeed if it manages to talk to *anyone*), and
-			// then we can clear the cache here only for retriable errors.
-			// It amounts to the same situations though.
+			// In the first case, we update the leader cache only.
 			if rpcErr != nil {
```

Comment:
`rpcErr` looks redundant, it's never used in any context where its value is not the same as `err`'s